 # Marouane Benabdelkader 

# Importation des Bibliothèques
- Tout d'abord, nous importons les librairies nécessaires pour notre script:

In [1]:
import os
import numpy as np
import csv

# Question 1
## Fonction `getData`

La fonction `getData` est utilisée pour charger les données depuis un fichier CSV en utilisant la bibliothèque numpy.

### Paramètres :

- `path` : `str`  
  Le chemin d'accès au fichier CSV à charger.

- `delem` : `str` (par défaut `,`)  
  Le délimiteur utilisé pour séparer les valeurs dans le fichier CSV.

### Retour :

- `np.ndarray`  
  Un tableau numpy contenant les données chargées du fichier CSV.

In [2]:
def getData(path: str, delem=',') -> np.ndarray:
    data = np.genfromtxt(path, delimiter=delem,
                            skip_header=1, dtype=None, encoding='utf-8')
    return data

## Fonction `reformat_data`

Transforme un tableau numpy de tuples en un tableau 2D et transpose les dimensions.

### Paramètres :

- `data` : `np.ndarray`  
  Tableau numpy de tuples.

### Retour :

- `np.ndarray`  
  Tableau numpy 2D avec les dimensions transposées.


In [3]:
def reformat_data(data: np.ndarray) -> np.ndarray:
    # Use zip and the * operator to unpack the tuples into columns
    columns = zip(*data)
    # Convert the columns back into a 2D array
    array_2d = np.array(list(columns))
    # Transpose the array to switch rows and columns back to their original structure
    array_2d = array_2d.T
    return array_2d

## Chargement et reformatage des données

Après avoir défini les fonctions nécessaires, nous exécutons les étapes suivantes pour charger et reformatage les données :

In [4]:
data = getData('./STATISTIQUES_EQUIPE_LIGUE1.csv', delem=';')
data = reformat_data(data)
print(data.shape)
victoires = data[1, 1].astype(int)
nuls = data[1, 2].astype(int)
defaites = data[1, 3].astype(int)
    # calcul de nombre de match total
match_total = victoires + nuls + defaites

(20, 16)


# Question 2
## Fonction `moyen_but`

Calcule la moyenne de buts par match pour l'ensemble des équipes.

### Paramètres :

- `data` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes.

### Retour :

- `float`  
  La moyenne de buts par match calculée pour toutes les équipes.


In [5]:
def moyen_but(data: np.ndarray) -> float:
    # Convertir les colonne en entiers
    butspour = data[:, 4].astype(int).sum()
    total_equipe = len(data)
    moyenne_but = butspour / ((match_total * total_equipe)/2)
    return moyenne_but

In [6]:
moyen_goals = moyen_but(data)
print("La moyenne des but des matchs est: ", moyen_goals)

La moyenne des but des matchs est:  2.96


# Question 3

## Fonction `moyen_points`

Calcule la moyenne de points pour l'ensemble des équipes basée sur leurs victoires et matchs nuls.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes.

### Retour :

- `float`  
  La moyenne de points calculée pour toutes les équipes.





In [7]:
def moyen_points(donnees: np.ndarray) -> float:
    """Retourne le nombre de point"""
    # Convertir la colonne Victoire en entiers
    victoires = donnees[:, 1].astype(int)
    # Convertir la colonne Nuls en entiers
    nuls = donnees[:, 2].astype(int)
    moyen = (victoires * 3 + nuls).sum() / 20
    return moyen

In [8]:
print("La moyenne des points est: ", moyen_points(data))

La moyenne des points est:  20.5


# Question 4
## Fonction `grand_nombre_victoire`

Identifie l'équipe avec le plus grand nombre de victoires.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes.

### Retour :

- `str`  
  Le nom de l'équipe ayant le plus grand nombre de victoires.








In [9]:
def grand_nombre_victoire(donnees: np.ndarray) -> str:
    """Retourne l'equipe avec le plus grand nombre de victoire"""
    victoires = donnees[:, 1].astype(int)
    max_victoire = np.argmax(victoires)
    equipe_vict = donnees[max_victoire, 0]
    return equipe_vict

In [10]:
equipe_vict = grand_nombre_victoire(data)
print("L'equipe avec le plus grand nombre de victoire est: ", equipe_vict)

L'equipe avec le plus grand nombre de victoire est:  PSG


# Question 5
## Fonction `grand_nombre_but`

Identifie l'équipe avec le plus grand nombre de buts marqués.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes, y compris les buts marqués.

### Retour :

- `str`  
  Le nom de l'équipe ayant marqué le plus grand nombre de buts.


In [11]:

def grand_nombre_but(donnees: np.ndarray) -> str:
    """Retourne l'equipe avec le plus grand nombre de but"""
    butspour = donnees[:, 4].astype(int)
    max_but = np.argmax(butspour)
    equipe_max_but = donnees[max_but, 0]
    return equipe_max_but

In [12]:
equipe_max_but = grand_nombre_but(data)
print("L'equipe avec le plus grand nombre de but est: ", equipe_max_but)

L'equipe avec le plus grand nombre de but est:  PSG


# Question 6
## Fonction `max_hors_jeux`

Identifie l'équipe avec le plus grand nombre de hors-jeux signalés.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes, y compris les occurrences de hors-jeux.

### Retour :

- `str`  
  Le nom de l'équipe avec le plus grand nombre de hors-jeux.


In [13]:
def max_hors_jeux(donnees: np.ndarray) -> str:
    """Retourne l'equipe avec le plus grand nombre de hors-jeux"""
    hors_jeux = donnees[:, 9].astype(int)
    max_hors_jeux = np.argmax(hors_jeux)
    equipe_hors_jeux = donnees[max_hors_jeux, 0]
    return equipe_hors_jeux

In [14]:
equipe_max_hrjs = max_hors_jeux(data)
print("L'equipe avec le plus grand nombre de hors-jeux est: ", equipe_max_hrjs)

L'equipe avec le plus grand nombre de hors-jeux est:  PSG


# Question 7
## Fonction `meilleur_defense_attaque`

Identifie les équipes avec la meilleure attaque et la meilleure défense.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipe

### Retour :

- `dict`  
  Un dictionnaire avec deux clés : "meilleur_attaque" et "meilleur_defense", chacune associée à une liste d'équipes.

### Description :

Cette méthode permet de gérer les cas où plusieurs équipes ont le même nombre maximal de buts marqués ou minimal de buts encaissés, en les incluant toutes dans le résultat.


In [15]:
def meilleur_defense_attaque(donnees: np.ndarray) -> dict:
    """Return the team with the best defense and the team with the best offense."""
    butspour = donnees[:, 4].astype(int)
    butscontre = donnees[:, 5].astype(int)

    # Find the teams with the maximum goals for and minimum goals against
    meilleur_attaque = donnees[butspour == np.amax(butspour), 0]
    meilleur_defense = donnees[butscontre == np.amin(butscontre), 0]

    return {"meilleur_attaque": meilleur_attaque.tolist(), "meilleur_defense": meilleur_defense.tolist()}

In [16]:
meilleur_defense_attaque = meilleur_defense_attaque(data)
print("L'equipe avec la meilleure defense est: ", meilleur_defense_attaque["meilleur_defense"])
print("L'equipe avec la meilleure attaque est: ", meilleur_defense_attaque["meilleur_attaque"])

L'equipe avec la meilleure defense est:  ['PSG']
L'equipe avec la meilleure attaque est:  ['PSG']


# Question 8
## Fonction `equipes_deficit_buts`

Identifie les équipes dont le nombre de buts marqués est inférieur au nombre de buts encaissés.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes.

### Retour :

- `np.ndarray`  
  Un tableau numpy contenant les équipes en déficit de buts.


In [17]:
def equipes_deficit_buts(donnees: np.ndarray) -> np.ndarray:
    """Trouve les équipes dont le nombre de buts marqués est inférieur au nombre de buts encaissés."""
    butspour = donnees[:, 4].astype(int)
    butcontre = donnees[:, 5].astype(int)
    equipes_deficit_but = donnees[butspour < butcontre]
    return equipes_deficit_but

In [18]:
equipes_deficit_but = equipes_deficit_buts(data)
print("Les equipes avec un deficit de buts sont: ", equipes_deficit_but[:, 0])

Les equipes avec un deficit de buts sont:  ['Ajaccio' 'Angers' 'Auxerre' 'Brest' 'Clermont' 'Montpellier' 'Nantes'
 'Nice' 'Reims' 'Strasbourg' 'Toulouse' 'Troyes']


# Question 9 
## Fonction `but_difference`

Calcule la différence entre les buts marqués et les buts encaissés pour chaque équipe.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes, y compris les buts marqués et encaissés.

### Retour :

- `np.ndarray`  
  Un tableau numpy représentant la différence de buts pour chaque équipe.


In [19]:
def but_difference(donnees: np.ndarray) -> np.ndarray:
    """calculer la difference entre buts marques et buts encaisses"""
    butspour = donnees[:, 4].astype(int)
    butcontre = donnees[:, 5].astype(int)
    difference = butspour - butcontre
    return difference

In [20]:
but_diff = but_difference(data)
for i, equipe in enumerate(data[:, 0]):
        print(f"Equipe {equipe}: {but_diff[i]} difference")

Equipe Ajaccio: -10 difference
Equipe Angers: -19 difference
Equipe Auxerre: -17 difference
Equipe Brest: -12 difference
Equipe Clermont: -4 difference
Equipe Lens: 16 difference
Equipe Lille: 4 difference
Equipe Lorient: 5 difference
Equipe Lyon: 5 difference
Equipe Marseille: 13 difference
Equipe Monaco: 8 difference
Equipe Montpellier: -5 difference
Equipe Nantes: -7 difference
Equipe Nice: -2 difference
Equipe PSG: 34 difference
Equipe Reims: -5 difference
Equipe Rennes: 17 difference
Equipe Strasbourg: -9 difference
Equipe Toulouse: -7 difference
Equipe Troyes: -5 difference


# Question 10
## Fonction `crenaux_but_pourcentage`

Calcule le pourcentage de buts marqués dans chaque créneau horaire pour chaque club.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes, y compris les buts marqués par créneau horaire.

### Retour :

- `np.ndarray`  
  Un tableau numpy représentant le pourcentage de buts marqués dans chaque créneau horaire pour chaque équipe.


In [21]:
def crenaux_but_pourcentage(donnees: np.ndarray) -> np.ndarray:
    """Calcule le pourcentage de buts marqués dans chaque créneau horaire pour chaque club."""
    but_par_crenaux = donnees[:, 10:16].astype(int)
    pourcentage = np.around((but_par_crenaux / match_total) * 100, 2)
    return pourcentage

In [22]:
pourcentage_but_crenaux = crenaux_but_pourcentage(data)
print(pourcentage_but_crenaux)

[[ 0.    6.67 33.33 20.   20.   13.33]
 [13.33  6.67 13.33 13.33 26.67 26.67]
 [20.    6.67 13.33  6.67 20.   26.67]
 [20.   13.33  6.67 20.   33.33 13.33]
 [20.    0.    6.67 40.   26.67 33.33]
 [ 6.67 26.67 13.33 33.33 40.   53.33]
 [26.67 26.67 46.67 20.   26.67 33.33]
 [33.33 26.67 33.33 46.67 13.33 20.  ]
 [20.   26.67 20.   20.   33.33 33.33]
 [33.33  6.67 53.33 26.67 20.   33.33]
 [20.   13.33 46.67 46.67 46.67 20.  ]
 [53.33 26.67 13.33  6.67 20.   40.  ]
 [ 6.67  6.67 13.33 20.   26.67 40.  ]
 [ 6.67 20.   20.   20.   20.   13.33]
 [33.33 46.67 46.67 46.67 33.33 80.  ]
 [ 6.67 33.33  6.67  6.67 13.33 40.  ]
 [13.33 33.33 20.   53.33 20.   66.67]
 [13.33 20.   13.33 13.33 26.67 26.67]
 [13.33 20.   40.   33.33 20.    6.67]
 [26.67 26.67 20.   33.33 13.33 53.33]]


# Question 11
## Fonction `but_par_match`

Calcule le nombre moyen de buts marqués par match pour chaque club.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes.

### Retour :

- `np.ndarray`  
  Un tableau numpy représentant la moyenne de buts marqués par match pour chaque équipe.

---

## Fonction `but_encaisse_par_match`

Calcule le nombre moyen de buts encaissés par match pour chaque club.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes.

### Retour :

- `np.ndarray`  
  Un tableau numpy représentant la moyenne de buts encaissés par match pour chaque équipe.

---

## Fonction `calcul_points`

Calcule le nombre total de points gagnés par chaque club.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes.

### Retour :

- `np.ndarray`  
  Un tableau numpy représentant le nombre total de points pour chaque équipe.

---

## Fonction `join_tables`

Joint plusieurs tableaux de statistiques pour chaque club.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes.

### Retour :

- `np.ndarray`  
  Un tableau numpy combinant diverses statistiques telles que les points, les victoires, les défaites, les buts marqués et encaissés, et les statistiques par créneau horaire pour chaque équipe.


In [23]:
def but_par_match(donnees: np.ndarray)-> np.ndarray:
    """Calcule le nombre de buts marqués par match pour chaque club."""
    butspour = donnees[:, 4].astype(int)
    buts_par_match = np.around(butspour / match_total, 2)
    return buts_par_match
def but_encaisse_par_match(donnees: np.ndarray)-> np.ndarray:
    """Calcule le nombre de buts encaissés par match pour chaque club."""
    butscontre = donnees[:, 5].astype(int)
    buts_encaisses_par_match = np.around(butscontre / match_total, 2)
    return buts_encaisses_par_match
def calcul_points(donnees: np.ndarray)-> np.ndarray:
    """Calcule le nombre de points gagnés par match pour chaque club."""
    victoires = donnees[:, 1].astype(int)
    nuls = donnees[:, 2].astype(int)
    points = (victoires * 3 + nuls) 
    return points
def join_tables(donnees: np.ndarray) -> np.ndarray:
    """Joins the goals scored and goals conceded tables together."""
    club = donnees[:, 0].reshape(20,1)
    points_equip = calcul_points(donnees).reshape(20,1)
    victoires = donnees[:, 1].astype(int).reshape(20,1)
    nuls = donnees[:, 2].astype(int).reshape(20,1)
    defaites = donnees[:, 3].astype(int).reshape(20,1)
    butspour = donnees[:, 4].reshape(20, 1)
    butcontre = donnees[:, 5].reshape(20, 1)
    horjeux = donnees[:, 9].reshape(20, 1)
    but_diff = but_difference(donnees).reshape(20, 1)
    buts_par_match = but_par_match(donnees).reshape(20, 1)
    buts_encaisses_par_match = but_encaisse_par_match(donnees).reshape(20, 1)
    pourcentage_but_crenaux = crenaux_but_pourcentage(donnees)
    # Join the tables together
    table = np.hstack((club, points_equip, victoires, nuls, defaites, butspour, butcontre, but_diff,horjeux, buts_par_match, buts_encaisses_par_match, pourcentage_but_crenaux))
    return table

   

In [24]:
join_tables = join_tables(data)
print(join_tables)

[['Ajaccio' '12' '3' '3' '9' '14' '24' '-10' '23' '0.93' '1.6' '0.0'
  '6.67' '33.33' '20.0' '20.0' '13.33']
 ['Angers' '8' '2' '2' '11' '15' '34' '-19' '17' '1.0' '2.27' '13.33'
  '6.67' '13.33' '13.33' '26.67' '26.67']
 ['Auxerre' '13' '3' '4' '8' '14' '31' '-17' '23' '0.93' '2.07' '20.0'
  '6.67' '13.33' '6.67' '20.0' '26.67']
 ['Brest' '13' '3' '4' '8' '16' '28' '-12' '26' '1.07' '1.87' '20.0'
  '13.33' '6.67' '20.0' '33.33' '13.33']
 ['Clermont' '19' '5' '4' '6' '19' '23' '-4' '25' '1.27' '1.53' '20.0'
  '0.0' '6.67' '40.0' '26.67' '33.33']
 ['Lens' '36' '11' '3' '1' '26' '10' '16' '19' '1.73' '0.67' '6.67'
  '26.67' '13.33' '33.33' '40.0' '53.33']
 ['Lille' '26' '8' '2' '5' '27' '23' '4' '24' '1.8' '1.53' '26.67'
  '26.67' '46.67' '20.0' '26.67' '33.33']
 ['Lorient' '28' '8' '4' '3' '26' '21' '5' '19' '1.73' '1.4' '33.33'
  '26.67' '33.33' '46.67' '13.33' '20.0']
 ['Lyon' '21' '6' '3' '6' '23' '18' '5' '23' '1.53' '1.2' '20.0' '26.67'
  '20.0' '20.0' '33.33' '33.33']
 ['Marseille

# Question 12
## Fonction `sort_by_points`

Trie les équipes par leur nombre de points, de la plus haute à la plus basse.

### Paramètres :

- `joined_table` : `np.ndarray`  
  Tableau numpy combinant diverses statistiques des équipes.

### Retour :

- `np.ndarray`  
  Le tableau numpy trié par points.


In [25]:
def sort_by_points(joined_table: np.ndarray) -> np.ndarray:
    """trier les equipes par points"""
    points = joined_table[:, 1].astype(float)
    table_sorter = joined_table[points.argsort()[::-1]]
    return table_sorter

In [26]:
sorted_table = sort_by_points(join_tables)
print(sorted_table)
print(sorted_table.shape)

[['PSG' '41' '13' '2' '0' '43' '9' '34' '36' '2.87' '0.6' '33.33' '46.67'
  '46.67' '46.67' '33.33' '80.0']
 ['Lens' '36' '11' '3' '1' '26' '10' '16' '19' '1.73' '0.67' '6.67'
  '26.67' '13.33' '33.33' '40.0' '53.33']
 ['Rennes' '31' '9' '4' '2' '31' '14' '17' '24' '2.07' '0.93' '13.33'
  '33.33' '20.0' '53.33' '20.0' '66.67']
 ['Marseille' '30' '9' '3' '3' '26' '13' '13' '20' '1.73' '0.87' '33.33'
  '6.67' '53.33' '26.67' '20.0' '33.33']
 ['Lorient' '28' '8' '4' '3' '26' '21' '5' '19' '1.73' '1.4' '33.33'
  '26.67' '33.33' '46.67' '13.33' '20.0']
 ['Monaco' '27' '8' '3' '4' '29' '21' '8' '26' '1.93' '1.4' '20.0'
  '13.33' '46.67' '46.67' '46.67' '20.0']
 ['Lille' '26' '8' '2' '5' '27' '23' '4' '24' '1.8' '1.53' '26.67'
  '26.67' '46.67' '20.0' '26.67' '33.33']
 ['Lyon' '21' '6' '3' '6' '23' '18' '5' '23' '1.53' '1.2' '20.0' '26.67'
  '20.0' '20.0' '33.33' '33.33']
 ['Nice' '20' '5' '5' '5' '15' '17' '-2' '27' '1.0' '1.13' '6.67' '20.0'
  '20.0' '20.0' '20.0' '13.33']
 ['Clermont' '19'

# Question 13
## Fonction `correlation_matrice`

Calcule la matrice de corrélation entre les différentes statistiques des équipes, y compris les buts marqués et encaissés.

### Paramètres :

- `donnees` : `np.ndarray`  
  Tableau numpy contenant les statistiques des équipes.

### Retour :

- `np.ndarray`  
  La matrice de corrélation représentant les relations statistiques entre les différentes colonnes de données.


In [27]:
def correlation_matrice(donnees: np.ndarray) -> np.ndarray:
    """Calcule la matrice de corrélation entre les colonnes de buts marqués et buts encaissés."""
    correlation_matrix = np.corrcoef(sort_by_points(data)[:, 1:].astype(float))
    return correlation_matrix

In [28]:
correlation = correlation_matrice(sorted_table)
print(correlation)

[[1.         0.99989645 0.99983912 0.99993244 0.99978084 0.99993432
  0.99990989 0.99996803 0.99998456 0.99993463 0.99973988 0.99983409
  0.99988416 0.99974223 0.99987898 0.99947944 0.99948517 0.99977352
  0.99980679 0.99980803]
 [0.99989645 1.         0.99999233 0.99999525 0.99997311 0.99998941
  0.99999634 0.99997518 0.99994609 0.99998744 0.99994951 0.99998186
  0.99998517 0.99995661 0.99998581 0.99983091 0.9998361  0.99997097
  0.99997406 0.99998032]
 [0.99983912 0.99999233 1.         0.9999787  0.99999196 0.99997348
  0.9999875  0.99994496 0.99990506 0.99996994 0.99997499 0.99999059
  0.99998069 0.99998039 0.9999836  0.99988885 0.99989484 0.9999892
  0.99998554 0.99999228]
 [0.99993244 0.99999525 0.9999787  1.         0.99995418 0.99999649
  0.99999623 0.99998973 0.99996954 0.99999477 0.99992906 0.99997182
  0.99998378 0.9999336  0.99998499 0.99978223 0.99978679 0.99995066
  0.9999599  0.99996462]
 [0.99978084 0.99997311 0.99999196 0.99995418 1.         0.99995064
  0.9999686  0.99

# Interprétation de la Matrice de Corrélation

La matrice de corrélation suivante présente les coefficients de corrélation entre différents champs statistiques d'une ligue de football :

- `club`
- `points_equip`
- `victoires`
- `nuls`
- `defaites`
- `butspour`
- `butcontre`
- `but_diff`
- `horjeux`
- `buts_par_match`
- `buts_encaisses_par_match`
- `pourcentage_but_crenaux`

Chaque cellule de la matrice représente le coefficient de corrélation entre les champs correspondants aux indices de la ligne et de la colonne.

## Observations Clés

- Tous les coefficients sont extrêmement proches de **1.0**, ce qui indique des corrélations positives très fortes entre les variables.
- Cela suggère que les équipes avec un plus grand nombre de **victoires**, **nuls**, et **points** ont tendance à avoir un nombre élevé de **buts marqués** (`butspour`) et une meilleure **différence de buts** (`but_diff`).
- De manière similaire, il semble que les **horaires de jeu** (`horjeux`), **buts par match**, et **buts encaissés par match** sont fortement liés aux performances globales des clubs.
- Ces résultats peuvent indiquer que les équipes les plus performantes sont celles qui non seulement marquent beaucoup de buts mais aussi maintiennent une défense solide, comme le montre la forte corrélation entre `butspour`, `butcontre`, et `points_equip`.


## Exemple de Valeurs de la Matrice

Voici un extrait de la matrice de corrélation pour les premières variables :

|                       | points_equip | victoires | nuls   | defaites | ... |
|-----------------------|--------------|-----------|--------|----------|-----|
| **points_equip**      | 1.0000       | 0.9999    | 0.9998 | 0.9999   | ... |
| **victoires**         | 0.9999       | 1.0000    | 0.9999 | 0.9999   | ... |
| **nuls**              | 0.9998       | 0.9999    | 1.0000 | 0.9999   | ... |
| **defaites**          | 0.9999       | 0.9999    | 0.9999 | 1.0000   | ... |
| ...                   | ...          | ...       | ...    | ...      | ... |



# Question 14
## Fonction `write_csv_file`

Enregistre les données statistiques des équipes dans un nouveau fichier CSV.

### Paramètres :

- `joined_table` : `np.ndarray`  
  Tableau numpy contenant les statistiques combinées des équipes.

### Fonctionnalité :

- La fonction utilise `np.savetxt` pour écrire les données dans un fichier CSV nommé "STATISTIQUES_EQUIPE_LIGUE1VersionNew.csv".
- Le fichier CSV est créé avec un en-tête spécifiant les colonnes : club, points, victoire, nul, defaite, butspour, butcontre, difference, hors_jeux, but_par_match, but_encraise, ainsi que les tranches horaires pour les buts.
- Un message est imprimé à l'écran une fois le fichier CSV créé avec succès.


In [29]:
def write_csv_file(joined_table):
    """Ecrire les donnees dans un nouveau csv file"""
    # if "STATISTIQUES_EQUIPE_LIGUE1VersionNew.csv" in os.listdir():
    #     os.remove("STATISTIQUES_EQUIPE_LIGUE1VersionNew.csv")
    np.savetxt("STATISTIQUES_EQUIPE_LIGUE1VersionNew.csv", joined_table, delimiter=",", fmt="%s",
               header="club, points, victoire, nul, defaite, butspour, butcontre, difference, hors_jeux, but_par_match, but_encraise,0-15,16-30,31-45,46-60,61-75,76-90")
    print("Fichier csv cree avec succes")

In [30]:
write_csv_file(sorted_table)

Fichier csv cree avec succes
